File: recursion/factorial-number.cpp
#include <iostream>
#include <string>

using namespace std;

// function to rekursif
int factorial(int n)
{
  if (n <= 1)
  {
    return 1;
  }
  else
  {
    // this is recursion arya call itself inside its body
    return n * factorial(n - 1);
  }
}

int main()
{
  int userInput;

  while (true)
  {
    // for user to input the number
    cout << "\nEnter your number: " << endl;
    cin >> userInput;

    // just create lines
    cout << string(25, '-') << endl;

    // connected to factorial function
    int cek_factorial = factorial(userInput);

    // to cek using if statement
    if (userInput <= 1)
    {
      cout << "Invalid input! \nYou have exited the program" << endl;
      break;
    }
    else
    {
      cout << "Faktorial from " << userInput << " is " << cek_factorial << endl;
    }
  }

  return 0;
}

File: recursion/factorial-number.py
def faktorial(n):
    if n <= 1:
        return 1
    else:
        return n * faktorial(n - 1)

while True:
    # Untuk pengguna menginput angkanya
    user_input = int(input("\nEnter the number: "))

    print("-" * 40)

    cek_factorial = faktorial(user_input)
    if user_input <= 1:
        print("Invalid input!\nYou've exited the program")
        break
    else:
        print(f"Faktorial dari {user_input} adalah: {cek_factorial}")
